src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1055 @@
+    private static String SYSTEM = "rest/system";
+    private static String REBOOT = "rest/system/reboot";
+    private static String COLLABORATION = "rest/collaboration";
+    private static String MICROPHONES = "rest/audio/microphones";
+    private static String APPS = "rest/system/apps";
+    private static String SESSIONS = "rest/current/session/sessions";
+    private static String URL_TEMPLATE = "%s://%s/%s";
+
+    private final ReentrantLock controlOperationsLock = new ReentrantLock();
+    private long latestControlTimestamp;
+    private ExtendedStatistics localStatistics;
+    private EndpointStatistics localEndpointStatistics;
+
+    private String sessionId;
+    private static final int CALL_RATE = 1920;
+    private static final int MAX_STATUS_POLL_ATTEMPT = 5;
+
+    /* Grace period for device reboot action. It takes about 3 minutes for the device to get fully
+     * functional after reboot is triggered */
+    private static final int REBOOT_GRACE_PERIOD_MS = 200000;
+
+    /**
+     * Instantiate and object with trustAllCertificates set to 'true' for being able to
+     * communicate with the devices with invalid certificates
+     */
+    public PolycomVideoOS() {
+        setTrustAllCertificates(true);
+    }
+
+    @Override
+    protected void internalInit() throws Exception {
+        super.internalInit();
+    }
+
+    @Override
+    public String dial(DialDevice dialDevice) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Dialing using dial string: " + dialDevice.getDialString());
+        }
+        ObjectNode request = JsonNodeFactory.instance.objectNode();
+        Integer callSpeed = dialDevice.getCallSpeed();
+
+        request.put("address", dialDevice.getDialString());
+        request.put("rate", (callSpeed != null && callSpeed > 0) ? callSpeed : CALL_RATE);
+
+        Protocol protocol = dialDevice.getProtocol();
+        if (protocol != null) {
+            request.put("dialType", protocol.name());
+        }
+        ArrayNode response = doPost(CONFERENCE, request, ArrayNode.class);
+        if (response == null) {
+            return null;
+        }
+
+        for (int i = 0; i < MAX_STATUS_POLL_ATTEMPT; i++) {
+            JsonNode meetingInfo = doGet(buildHttpUrl(getJsonProperty(response.get(0), "href", String.class)), JsonNode.class);
+            if (meetingInfo != null) {
+                String conferenceId = getJsonProperty(meetingInfo, "parentConfId", String.class);
+                if (null != conferenceId) {
+                    String remoteAddress = getJsonProperty(meetingInfo, "address", String.class);
+                    if (!StringUtils.isNullOrEmpty(remoteAddress) && remoteAddress.trim().equals(dialDevice.getDialString().trim())) {
+                        return conferenceId;
+                    }
+                }
+            }
+            Thread.sleep(1000);
+        }
+        return null;

```

Comment:
I have checked other communicators that use CallController in symphony-communicator and used the similar approach with both retry mechanism and returning null at the end (like in PolycomVSX, CiscoCTS, CiscoX for instance, but this seems to be a common way for such communicators).

Reply:
Not 100% sure that's correct approach, but ok. But another problems there is what it is returning. It is not the conference id it is supposed to return, it is a call id. Call id  is unique for each call, if there are multiple calls connected to the same conference, each of them has its own call id. And if the same device connects to the conference, then disconnects and then connects again, the call ids will be different. Depending on how destination deals with calls, the call id can be in different format. Sometimes it is UUID, sometimes it is number which is being incremented by remote side, sometimes it is just a string. Most of devices (or MCUs) support call ids because it is needed on the remote side to identify the call. But, there are few old ones which do not, not 100% sure what those communicators do.
In any case, returning conference id here does not look correct.

Reply:
Basically, in VideoOS - conferenceId is an ID of a conference call that is created whenever the connection is established, while the connectionId is an identifier of a specific conference's connection, that contains specific device's information:  communication protocol, address etc, so if it would be Zoom Meeting, for instance, that VideoOS device is connecting to, the conference will be created and within the conference, there would be a connection available, containing information about the Zoom Meeting the device is connected to - protocol, address etc. 

So from the perspective of communication with the API - in order to start a call we need to interact with "/rest/conferences" resource, which made me think that from this point we need to interact with the call accordingly, as it contains rather general information about the connection and in order to retrieve any information about any particular connection (which is a property of a conference) - we need to have a conference ID available as well. 

However, when the _/rest/conferences_ request is posted, the response would contain URI of the connection (e.g /rest/conferences/<confID>/connections/<connID>), which then will allow us to get information about the conference. Alternatively, this URI may be used as an identifier for the call, so we will still have all the conference information needed within the reach. 

Which one sounds more logical to you?

Reply:
I do not know much about VideoOS. What is "conference" in VideoOS word? Does VideoOS creates a local conference which can have multiple calls in it? What would happen if multiple devices connect to that conference? (Is it possible?)
Or it is just a terminology for a call? (Naming might be confusing). And multiple connections there just represent different channels? (signaling, media, etc). Without knowing how it works I cannot really advise

Reply:
Yes, the latter one is rather what it is, conference is a different word for a call. Connections are participants, so if it is a p2p call - a conference will be created and contain general call information, and connection will contain a list of devices that are connected to the call (in a single participant call it's gonna be 1 connection, the device itself supports blast dial - a conference with multiple participants, but currently on a dial() call regular p2p conference will be created).

Reply:
OK, so why is that not documented? Is the expectation that whoever might work on this adapter in the future will have to reverse engineer or that knowledge ;)? The rule of thumb is that if it is not obvious from the code, it must be documented either using inline comments or JavaDocs (whatever is more appropriate in each particular case). Since I'm asking that many question, it is definitely not obvious -> thus must be documented.

Reply:
Ok, thank you, adding javadocs + referencing a VideoOS documentation (basically this API specific is covered by the VideoOS doc but I agree that this should be rather explicit)

Reply:
I still do not understand whey do we return a null here and do not see any new documentation about it.

Reply:
Added description of a null case.

Reply:
Maksim, I'm sorry but we really need to do the right thing here.
If you dial, and APIs support callId in any format (even if it is a conferenceId here, another discussion), there are only two possible cases:
1) Success - dial succeeded and call id is returned
2) Failure - everything else.
There is no third case. The only time null can be returned (and documented) if call id is not supported by APIs.
So, in you case, you have to throw an error if you end up with not having call id.

Reply:
Okay, I see, thanks!
I've checked a couple of other communicators, such as Cisco(CTS, X) and Polycom (HDR, VSX) while making this desicion, and these've been following the same approach, this is the reason of the confusion. 
Thank you for the clarification